airflow/providers/google/cloud/triggers/dataproc.py

```diff
@@ -140,24 +142,97 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
                 "gcp_conn_id": self.gcp_conn_id,
                 "impersonation_chain": self.impersonation_chain,
                 "polling_interval_seconds": self.polling_interval_seconds,
+                "delete_on_error": self.delete_on_error,
             },
         )
 
     async def run(self) -> AsyncIterator[TriggerEvent]:
-        while True:
-            cluster = await self.get_async_hook().get_cluster(
-                project_id=self.project_id, region=self.region, cluster_name=self.cluster_name
+        """Run the trigger."""
+        try:
+            while True:
+                cluster = await self.fetch_cluster_status()
+                if self.check_cluster_state(cluster.status.state):
+                    if cluster.status.state == ClusterStatus.State.ERROR:
```

Comment:
We already checked the cluster state in the above method `check_cluster_state` whether it belongs to ERROR. Can we avoid the duplicate if conditional check here for the ERROR state?

Reply:
I think the idea here is if the error is there - gather_diagnostics_and_maybe_delete() should be awaited else ```yield TriggerEvent(
                            {
                                "cluster_name": self.cluster_name,
                                "cluster_state": cluster.status.state,
                                "cluster": cluster,
                            }
                        )``` should be yield.